meson/libssl/meson.build

```diff
@@ -1,5 +1,18 @@
 opt_libssl = get_option('tls-libssl')
-dep_libssl = dependency('libssl', required: opt_libssl)
+opt_libssl_dir = get_option('tls-libssl-dir')
+
+
+if opt_libssl_dir != ''
+  dep_libssl = declare_dependency(
+    include_directories: [opt_libssl_dir / 'include'],
+  )
+  add_global_link_arguments(
+    ['-L' + opt_libssl_dir / 'lib', '-lssl'],
+    language: ['c', 'cpp'],
```

Comment:
I did not test this suggestion, sorry, but I was wondering if we could pass these via `declare_dependency`'s `link_args` and `link_with` instead? Because I'm worried we might otherwise be linking `libssl` in binaries/shared libraries that might not need it.

Reply:
Yes, that's better indeed, will fix.